arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,720 @@
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}/commitmentAssociations/{associationName}": {
+      "get": {
+        "tags": [
+          "CommitmentAssociations"
+        ],
+        "operationId": "CommitmentAssociations_Get",
+        "description": "Gets a commitment association.",
+        "parameters": [
+          {
+           "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "planName",
+            "description": "The association's parent commitment plan name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "associationName",
+            "description": "The commitment association name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommitmentAssociation"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}/commitmentAssociations": {
+      "get": {
+        "tags": [
+          "CommitmentAssociations"
+        ],
+        "operationId": "CommitmentAssociations_List",
+        "description": "Get all commitment associations for a parent commitment plan.",
+        "parameters": [
+          {
+           "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "planName",
+            "description": "The association's parent commitment plan name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "$skipToken",
+            "description": "Continuation token for pagination.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PaginatedResult[CommitmentAssociation]"
```

Comment:
This `PaginatedResult[CommitmentAssociation]` is a weird way to name a model class. A better name would be `PaginatedListResult` and that would also follow the standard convention.
/cc @kirthik @tbombach - One more rule to add in the linter - If Model.Name contains `/(.*)\[(.*)\]/ig` then it should be sensibly named as "$2ListResult".